libs/wingc/src/lib.rs

```diff
@@ -254,6 +257,91 @@ fn add_builtin(name: &str, typ: Type, scope: &mut Scope, types: &mut Types) {
 		.expect("Failed to add builtin");
 }
 
+/// The different phases of compilation, used for tracking compilation context
+/// for diagnostic purposes. Feel free to add new phases as needed.
+/// Try to make these end with 'ing' (building, parsing, etc.) so they'll fit in
+/// in the context of a diagnostic message.
+/// See `CompilationContext::set` for more information.
+#[derive(EnumString, Display, PartialEq, Clone, Copy)]
+#[strum(serialize_all = "kebab-case")]
```

Comment:
I wonder if we can avoid adding a new dependency by just implementing the FromStr trait on this type (shouldn't be that many lines of code)?